### PR TITLE
Isolate module

### PR DESCRIPTION
**Pull request by TylorS**
First attempt at isolation via a snabbdom hook, and no DOM manipulation.

### CODE REVIEW

**Comment by TylorS** [comment]
Should also be a fix for #29

File: dist/cycle-snabbdom.js [hunk 1]
```diff
@@ -2947,7 +2947,7 @@ function updateEventListeners(oldVnode, vnode) {
 module.exports = {create: updateEventListeners, update: updateEventListeners};
 
 },{"../is":34}],38:[function(require,module,exports){
-var raf = (typeof window !== `undefined` && window.requestAnimationFrame) || setTimeout;
+var raf = (typeof window !== 'undefined' && window.requestAnimationFrame) || setTimeout;
```
Comment (axefrog) [inline dist/cycle-snabbdom.js:2950]: For what it's worth, I've noticed Brian tests for undefined a lot via `foo === void 0`, which avoids having to perform a type check. Might be very slightly more efficient.

Comment (TylorS) [inline dist/cycle-snabbdom.js:2950]: That's actually snabbdom code that gets included with the dist/ build

**Comment by laszlokorte** [comment]
👍  I already thought about something similar. It's amazing to see you have already worked on this.
I would be great if in a next step we could attach each event listener to the isolation boundary element instead of to the dom driver's root element in order to achieve correct order of event propagation (https://github.com/cyclejs/core/issues/229)

**Comment by TylorS** [comment]
@lazlokorte that's actually occurring here :)

**Comment by TylorS** [comment]
Oops sorry I lied :(

**Comment by TylorS** [comment]
It will be really easy to achieve though, I'm already doing that for `.observable`. Not sure why I didn't carry that over to `.events()`.

File: dist/cycle-snabbdom.js (hunk 1 above)
Comment (axefrog) [inline dist/cycle-snabbdom.js:2950]: Yes, I am an idiot. Ignore my unintelligible utterances.

**Comment by TylorS** [comment]
@laszlokorte With the my latest commit it attaches event listeners to isolated components rather than the rootElement :)

Will also do the same for @cycle/dom - which hopefully soon can be released and I can deprecate this package

**Pull request merged**